matscipy/dislocation.py

```
@@ -3801,7 +3969,86 @@ def build_glide_quadrupoles(self, nims, invert_direction=False, glide_left=True,
+        Mask is needed in building kinks to ensure continuity of atomic basis
+
+        ref bulk: ase Atoms
+            Bulk structure to use to generate mask
+        direction: int
+            Direction of kink (+1 or -1)
+        precision: int
+            Scaled precision to use when determining a step size
+            Each step is self.alat * 10**_precision Ang
+            
+
+        returns a boolean mask, and a new cell
+
+        '''
+
+        if self.left_dislocation.__class__ == BCCEdge111barDislocation:
+            #raise RuntimeError("Kink Quadrupoles do not currently work for BCCEdge111barDislocation")
+            pass
+
+        bulk = ref_bulk.copy()
+        tilt_bulk = ref_bulk.copy()
+        cell = tilt_bulk.cell[:, :]
+        cell[2, 0] += direction * self.glide_distance
+
+        tilt_bulk.set_cell(cell, scale_atoms=False)
+        tilt_bulk.wrap()
+
+        p = ref_bulk.get_positions()
+
+        step_size = self.alat * 10**(-precision)
+
+        # Shift bulk if atoms too close to the border
+        if any(p[:, 2] < step_size):
+            p[:, 2] += step_size
+            tilt_bulk.set_positions(p)
+            tilt_bulk.wrap()
+            p = tilt_bulk.get_positions()
+
+    
+        # 1st, 2nd, & 3rd nearest neighbour distances
+        # for each crytal structure
+        neigh_dists = {
+        "bcc" : [0.88, 1.01, 1.42],
+        "fcc" : [0.72, 1.01, 1.23],
+        "diamond" : [0.44, 0.72, 0.84]
+        }
+        cutoffs = neigh_dists[self.crystalstructure]
+
+        bulk_coord = [np.min(coordination(bulk, cutoff * self.alat)) for cutoff in cutoffs]
+
+        full_mask = np.ones((len(tilt_bulk)), dtype=bool)
+        atom_heights = np.round(p, precision)[:, 2]
+
+        for i in range(int(cell[2, 2] / step_size)):
+            # Check if tilt_bulk has same 1st, 2nd, & 3rd neighbour coordination
+            # as normal bulk
+            # TODO: This currently does not work for BCCEdge111barDislocation
```

can you raise an exception?